test/unit/helpers/cache.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+var chai = require('chai');
+var expect = require('chai').expect;
+var sinon = require('sinon');
+var redis = require('redis');
+
+var cache = require('../../../helpers/cache');
+
+describe('cache', function () {
+
+	it('is not enabled', function () {
+		cache.connect(false, {}, {}, function (err, result) {
+			expect(result.cacheEnabled).to.eq( false );
+			expect(result.client).to.be.null;
+		});
+	});
+
+	it('is enabled redis client callback ready', function () {
+
+		var logger = {
+			'info': function (info) {expect(info).to.eq('App connected with redis server');}
+		};
+
+		var mockredisclient = {
+			'on': function (type, cb) {
+				if ( 'ready' === type ) {
+					cb();
+				}
+			}
+		};
+
+		var mockedRedis = sinon.mock(redis);
+		mockedRedis.expects('createClient').once().withArgs(sinon.match.any).returns(mockredisclient);
+		cache.connect(true, {}, logger, function (err, result) {
+			expect(result.cacheEnabled).to.eq( true );
+			expect(result.client).to.be.not.null;
+
+			mockedRedis.restore();
+			mockedRedis.verify();
+		});
+	});
+
+	it('is enabled redis client callback error', function () {
+
+		var logger = {
+			'error': function (message, err) {
+				expect(message).to.eq('Redis:');
+				expect(err + '').to.eq('Error: Cache Test Callback Error');
```

Comment:
You can do deep equality comparison with `.eql` function. So, please don't perform implicit conversion of Error to string value.

Reply:
Added. Made declare scope var ERROR.